### PR TITLE
#170947656 Send email verification link

### PR DESCRIPTION
**Pull request by Moise1**
#### What does this PR do?
- It sends an email verification link to the new user's email address. 
- Once the link is opened the user gets redirected back to the app.

#### Description of Task to be completed?
- Send a verification link to the newly signed up user's email address. 
- Save the verified user in the database immediately after verifying their account.

#### How should this be manually tested?
- run `git clone https://github.com/andela/knights-bn-backend.git` in your terminal
- run `cd knights-bn-backend` in your terminal
- run `git checkout  ft-verification-email-170947656`
- run `npm install` to install all packages. 
- run `npm run createTables` 
- run `npm run dev-start`
- copy this into your  **.env**  file: 
> `HOST_NAME: 'knights-bn-backend-staging.herokuapp.com'`

> `LOCAL_HOST='http://localhost:4000`'
- copy  the following in your Postman as a request body: 
```
{
     "firstName": "first_name",
     "lastName": "last_name",
     "gender": "Male",
    "passportNumber" :"1234567rv",
    "email": "your.email@example.com",
   "password": "password"
}
```
- Please use your valid email so you can be able to see the result after verifying your accound. 

- Hit the this endpoint: `http://localhost:4000/api/v1/auth/signup`

- Check your email address to see the message. 
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#170947656](https://www.pivotaltracker.com/story/show/170947656)

#### Postman' response body 

Screen Shot 2020-02-26 at 14 19 30

#### My email address 
Screen Shot 2020-02-26 at 14 20 19

### CODE REVIEW

**Comment by T2Wil** [comment]
hey @Moise1 , 
1. in the PR description you have not spedified a step of which route should be used to signup `http://localhost:4000/api/v1/auth/signup`
2. I think the request body you provided to run in postman is not well-formed and hence can't run... You may put it into JSON format.
3.  I am receiving `UnhandledPromiseRejectionWarning: Error: Unauthorized
    at Request._callback` error when I post the request and I am not receiving a verification link email.

image
image

**Comment by Moise1** [comment]
Hi @T2Wil , thanks for the feedback. I have applied it for points 1 & 2.  However, point 3 is probably because you don't' have the BN_API_KEY that I have used.

**Comment by EugeneMunya** [comment]
Resolve the conflict

**Comment by EugeneMunya** [comment]
Nice work @Moise1 but try to return well define error message when user is exist not like this
Screen Shot 2020-02-19 at 09 00 29

**Comment by Moise1** [comment]
@EugeneMunya the conflicts were resolved.

**Review by Niyonsengaeric — APPROVED** [review]

**Comment by T2Wil** [comment]
Hey @Moise1 , 
I was testing the endpoint, successfully received an email but the link is not working.

image

fix it, bro.

**Comment by Niyonsengaeric** [comment]
> Hey @Moise1 ,
> I was testing the endpoint, successfully received an email but the link is not working.
> 
> image
> 
> fix it, bro.

@T2Wil  have checked the sendGrid key you are using ?? because on my side it is working

**Comment by T2Wil** [comment]
> > Hey @Moise1 ,
> > I was testing the endpoint, successfully received an email but the link is not working.
> > image
> > fix it, bro.
> 
> @T2Wil have checked the sendGrid key you are using ?? because on my side it is working

Hey @Niyonsengaeric and @Moise1 , it is now working but I had to set the ```HOST_NAME=localhost:4000``` in my `.env`. 
@Moise1 , you should have added this .env variable as a requirement, however, you did a good job.

**Review by T2Wil — APPROVED** [review]

**Comment by Niyonsengaeric** [comment]
> > > Hey @Moise1 ,
> > > I was testing the endpoint, successfully received an email but the link is not working.
> > > image
> > > fix it, bro.
> > 
> > 
> > @T2Wil have checked the sendGrid key you are using ?? because on my side it is working
> 
> Hey @Niyonsengaeric and @Moise1 , it is now working but I had to set the `HOST_NAME=localhost:4000` in my `.env`.
> @Moise1 , you should have added this .env variable as a requirement, however, you did a good job.

@T2Wil  as I am seeing maybe you are missing some environment variables you can check on this picture:  like LOCAL_HOST, HOST_NAME
Screen Shot 2020-02-29 at 20 39 15

Thanks @Niyonsengaeric , actually that is what was missing but, I, fortunately found it and it worked!
nice job @Moise1

**Review by Kagorora — APPROVED** [review]

**Comment by EugeneMunya** [comment]
> > > Hey @Moise1 ,
> > > I was testing the endpoint, successfully received an email but the link is not working.
> > > image
> > > fix it, bro.
> > 
> > 
> > @T2Wil have checked the sendGrid key you are using ?? because on my side it is working
> 
> Hey @Niyonsengaeric and @Moise1 , it is now working but I had to set the `HOST_NAME=localhost:4000` in my `.env`.
> @Moise1 , you should have added this .env variable as a requirement, however, you did a good job.

@Moise1 tell everyone to configure HOST_NAME variable in .env in PR description not in a comment.

File: src/controllers/users.js [hunk 1]
```diff
@@ -31,48 +78,26 @@ export default class usersController {
...
+        email: newUser.email,
+        firstName: newUser.firstName,
+        lastName: newUser.lastName
+      }, process.env.SECRETKEY);
 
-      return res
-        .status(201)
-        .json({ message: 'user successfully created', token });
+      localStorage.setItem('token', token);
+      return res.status(201).json({ message: 'Your account is successfully created.', token });
```
Comment (EugeneMunya) [inline src/controllers/users.js]: I think to return token on verification processes is not necessary you can display only message

Comment (Moise1) [reply src/controllers/users.js]: @EugeneMunya , thanks for your feedback. It's solved now.

**Review by kagabof — APPROVED** [review]
Greate job @Moise1.

**Pull request merged**